template/src/bin/main.rs

```diff
@@ -67,12 +71,12 @@ fn main() -> ! {
     .unwrap();
     //ENDIF
 
-    let delay = Delay::new();
     loop {
         //IF option("defmt") || option("log")
         info!("Hello world!");
         //ENDIF
-        delay.delay_millis(500);
+        let delay_start = Instant::now();
+        while delay_start.elapsed() < Duration::from_millis(500) {}
     }
 
     // for inspiration have a look at the examples at https://github.com/esp-rs/esp-hal/tree/v0.23.1/examples/src/bin
```

Comment:
We should find a way to programatically derive this from the esp-hal version in the template's Cargo.toml. We can do that another time, for now we should just update this to the beta tag (when it exists). Don't forgot about the async template too.

Reply:
Yes, we should open an issue for it! Shall I update it to `esp-hal-v1.0.0-beta.0` or we keep the old naming here `v1.0.0-beta.0`?

Reply:
New tag will be `esp-hal-v1.0.0-beta.0`